### ISSUE TITLE
Command Line Monitor Numbers not Matching Windows Numbers

### ISSUE THREAD

**Issue by cerahmed** [xanderfrangos/twinkle-tray#628]
**Hello there,**

So I have been trying to write an AHK script that detects which monitor the mouse is currently hovering (returns the Windows monitor number found in Display Settings), then call Twinkle Tray command line to offset the brightness of that monitor.

The AHK works great in detecting the current monitor, however after some testing it seems that Twinkle Tray numbers monitors differently than how Windows does (not talking about the zero-indexing, the order itself is different).

<img width="358" alt="Screenshot 2023-07-31 050006" src="https://github.com/xanderfrangos/twinkle-tray/assets/21166481/41e8c96f-0793-48c4-abdd-bd4d665a1f83">

Note: Just while uploading the photo that I realized that it MIGHT be twinkle tray has the order in an opposite order!
But since I only have three monitors I can test with at the moment, I'm not sure if that's the case.

I'm after an AHK script that works no matter how many monitors I have.

_**TLDR; can you please explain the relation between Twinkle Tray monitor numbering vs. Windows Display Settings numbering?**_

All in all thanks for this great utility! Been rocking it for months and can't live without it!

**Comment by xanderfrangos** [comment]
Hi @cerahmed,

There's no relation I'm aware of between the "MonitorNum" value and Window's display index. Twinkle Tray's "MonitorNum" is based off of the detection order, and the list can potentially include inactive/duplicated displays. You'll probably need to manually map the display index from AHK with the "MonitorID" in Twinkle Tray.

**Comment by cerahmed** [comment]
Hi @xanderfrangos,

Thanks for your response.

> You'll probably need to manually map the display index from AHK with the "MonitorID" in Twinkle Tray.

That's indeed what I figured I should do to ensure consistency. Been researching since yesterday on how I could get such information and I finally managed to get it working!

Knowing the monitor ID under which the mouse pointer is at is really neat. Had to migrate to AHK v2 in order to run some scripts I found online, then did some tidying up and now I can change brightness using Win+Alt+Mouse Scroll up/down in increments of 5.

The AHK scripts I found uses Windows DLL API's that's easily accessible using C++, so maybe you could implement such a feature (e.g. a univeral hotkey that only controls the monitor where the pointer is currently at). For me better than having individual hotkeys for each monitor.

I'm sharing the AHK script that might be useful to someone.

[brightness.zip](https://github.com/xanderfrangos/twinkle-tray/files/12217316/brightness.zip)

**Comment by cerahmed** [comment]
Hi @xanderfrangos ,

My AHK code uses the `--Overlay` argument to nicely display the brightness levels while using the mouse shortcut.
However I noticed that the _**overlay is not set to display on top of other apps**_. If I have a full screen window, the overlay will be blocked by the app.

Anyway you could look into this?
Let me know if you want me to make a new issue for this one to better track it.

**Comment by xanderfrangos** [comment]
@cerahmed That's the intended behavior. The overlay does not display over windows that mark themselves as "always on top" (like when watching a video) or exclusive fullscreen applications (like many games). Forcing the overlay to display over those causes a number of unwanted side effects (input/window focus loss, games minimizing, performance issues, anti-cheat false flags, etc). An upcoming version of Twinkle Tray (v1.16.0) will allow for you to override that on a per-app basis, but the default behavior will not change.
